src/pint/plot_utils.py

```diff
@@ -176,3 +177,127 @@ def phaseogram_binned(
+    values = []
+    for item in chains.items():
+        keys.append(item[0]), values.append(item[1])
+
+    priors = []
+    initials = []
+    x_range = []
+    counts = []
+    for i in range(0, len(keys[:-1])):
+        values[i] = values[i][burnin:].flatten()
+        x_range.append(np.linspace(values[i].min(), values[i].max(), num=bins))
+        priors.append(getattr(model, keys[i]).prior.pdf(x_range[i]))
+        initials.append(
+            GaussianBoundedRV(loc=float(fitvals[i]), scale=float(fiterrs[i])).pdf(
+                x_range[i]
+            )
+        )
+        a, x = np.histogram(values[i], bins=bins)
+        counts.append(a)
+
+    fig, axs = plt.subplots(len(keys), figsize=(8, 11), constrained_layout=True)
+
+    for i, p in enumerate(keys):
+        if i != len(keys[:-1]):
+            axs[i].set_xlabel(
+                str(p)
+                + ": Max Post Value - "
+                + str("{:.9e}".format(maxpost_fitvals[i]))
+                + " ("
+                + str(getattr(model, p).units)
+                + ")"
+            )
+            axs[i].axvline(
+                fitvals[i] - maxpost_fitvals[i],
+                color="m",
+                linestyle="--",
+                label="Original Fit Value",
+            )
+            axs[i].axvline(
+                maxpost_fitvals[i] - maxpost_fitvals[i],
+                color="c",
+                linestyle="--",
+                label="Maximum Likelihood Value",
+            )
+            axs[i].axvline(
+                -2 * values[i].std(), color="b", linestyle="--", label="2 sigma"
+            )
+            axs[i].axvline(2 * values[i].std(), color="b", linestyle="--")
+            axs[i].hist(
+                values[i] - maxpost_fitvals[i], bins=bins, label="Samples",
+            )
+            if scale:
+                axs[i].plot(
+                    x_range[i] - maxpost_fitvals[i],
+                    initials[i] * counts[i].max() / initials[i].max(),
+                    label="Initial Gaussian Probability",
+                    color="r",
+                )
+                axs[i].plot(
+                    x_range[i] - maxpost_fitvals[i],
+                    priors[i] * counts[i].max() / priors[i].max(),
+                    label="Prior Probability",
+                    color="g",
+                )
+            else:
+                axs[i].plot(
+                    x_range[i] - maxpost_fitvals[i],
+                    initials[i],
+                    label="Initial Gaussian Probability",
+                    color="r",
+                )
+                axs[i].plot(
+                    x_range[i] - maxpost_fitvals[i],
+                    priors[i],
+                    label="Prior Probability",
+                    color="g",
+                )
+        else:
+            handles, labels = axs[0].get_legend_handles_labels()
+            axs[i].set_axis_off()
+            axs[i].legend(handles, labels)
+    if file:
```

Comment:
I think these are not a good idea. Users might want to add more things to the figure, so calling `plt.close()` is going to cause them problems. And they may want to specify extra arguments (`dpi` for example) when saving to a file, so let them save the thing if they want to.